pmm-app/src/pmm-settings/components/PlatformLogin/SignIn.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC } from 'react';
+import { Field, Form, FormRenderProps } from 'react-final-form';
+import { Button, useTheme } from '@grafana/ui';
+import validators from 'shared/components/helpers/validators';
+import { ButtonWithSpinner } from 'shared/components/Form';
+import { showErrorNotification, showSuccessNotification } from 'shared/components/helpers';
+import { Credentials, LoginFormProps } from './types';
+import { Messages } from './PlatformLogin.messages';
+import { InputFieldAdapter } from './FieldAdapters/FieldAdapters';
+import { getStyles } from './PlatformLogin.styles';
+import { PlatformLoginService } from './PlatformLogin.service';
+
+
+export const SignIn = ({ changeMode, getSettings }: LoginFormProps) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  const handleSignInFormSubmit = async (credentials: Credentials) => {
+    try {
+      await PlatformLoginService.signIn(credentials);
+
+      getSettings();
```

Comment:
It's not clear to me why we have to call `getSettings` in these pages. also there's a bit of prop drilling to pass the callback here.

Reply:
we need to call getSettings because we're not getting email from the backend anymore in answer, so after we sign in we need to re-request data to display login state